packages/vitest-pool-workers/test/helpers.ts

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable no-empty-pattern */
+import childProcess from "node:child_process";
+import events from "node:events";
+import fs from "node:fs/promises";
+import path from "node:path";
+import util from "node:util";
+import { stripAnsi } from "miniflare";
+import { test as baseTest, inject, vi } from "vitest";
+
+const debuglog = util.debuglog("vitest-pool-workers:test");
+
+export const minimalVitestConfig = `
+import { defineWorkersConfig } from "@cloudflare/vitest-pool-workers/config";
+export default defineWorkersConfig({
+	test: {
+		testTimeout: 90_000,
+		poolOptions: {
+			workers: {
+				singleWorker: true,
+				miniflare: {
+					compatibilityDate: "2024-01-01",
+					compatibilityFlags: ["nodejs_compat"],
+				},
+			},
+		},
+	}
+});
+`;
+
+export function waitFor<T>(callback: Parameters<typeof vi.waitFor<T>>[0]) {
+	// The default timeout of `vi.waitFor()` is only 1s, which is a little
+	// short for some of these tests, especially on Windows.
+	return vi.waitFor(callback, { timeout: 10_000, interval: 500 });
+}
+
+async function seed(root: string, files: Record<string, string>) {
+	for (const [name, contents] of Object.entries(files)) {
+		const filePath = path.resolve(root, name);
+		await fs.mkdir(path.dirname(filePath), { recursive: true });
+		await fs.writeFile(filePath, contents);
+	}
+}
+
+export interface Process {
+	readonly stdout: string;
+	readonly stderr: string;
+	readonly exitCode: Promise<number>;
+}
+
+function wrap(proc: childProcess.ChildProcess): Process {
+	let stdout = "";
+	let stderr = "";
+	proc.stdout?.setEncoding("utf8");
+	proc.stderr?.setEncoding("utf8");
+	proc.stdout?.on("data", (chunk) => {
+		if (debuglog.enabled) {
+			process.stdout.write(chunk);
+		}
+		stdout += chunk;
+	});
+	proc.stderr?.on("data", (chunk) => {
+		if (debuglog.enabled) {
+			process.stderr.write(chunk);
+		}
+		stderr += chunk;
+	});
+	const closePromise = events.once(proc, "close");
+	return {
+		get stdout() {
+			return stripAnsi(stdout);
+		},
+		get stderr() {
+			return stripAnsi(stderr);
+		},
+		get exitCode() {
+			return closePromise.then(([exitCode]) => exitCode ?? -1);
+		},
+	};
+}
+
+export const test = baseTest.extend<{
+	tmpPath: string;
+	seed: (files: Record<string, string>) => Promise<void>;
+	vitestRun: (options?: {
+		flags?: string[];
+		maxBuffer?: number;
+	}) => Promise<Process>;
+	vitestDev: (options?: { flags?: string[]; maxBuffer?: number }) => Process;
+}>({
+	// Fixture for creating a temporary directory
+	async tmpPath({}, use) {
+		const tmpPoolInstallationPath = inject("tmpPoolInstallationPath");
+		const tmpPathBase = path.join(tmpPoolInstallationPath, "test-");
+		const tmpPath = await fs.mkdtemp(tmpPathBase);
+		await use(tmpPath);
+		await fs.rm(tmpPath, { recursive: true, maxRetries: 10 });
```

Comment:
This type of cleanup occasionally fails on Windows in CI with `EBUSY`, so we may want to catch and ignore this erroring

Reply:
Interestingly I haven't seen any flakes of this in Windows testing. (And I was running them a lot today!)
So I suspect that using `maxRetries` is a good solution here.